### PR TITLE
Modularize `process_state_forecast.R` and add some basic tests

### PR DESCRIPTION
**Pull request by dylanhmorris**
Prerequisite for epiweekly predicted/actual plots but separable and therefore merging in now to prevent mega PRs

### CODE REVIEW

File: hewr/R/process_state_forecast.R [hunk 1]
```diff
@@ -1,21 +1,104 @@
...
+#' @param disease_name Name of the target disease.
+#' One of `"COVID-19"` or `"Influenza"`.
+#' @return The combined data, as a [`tibble`][tibble::tibble()].
+#' @export
+combine_training_and_eval_data <- function(train_dat,
+                                           eval_dat,
+                                           disease_name) {
+  combined_dat <- dplyr::bind_rows(
+    train_dat |>
+      dplyr::filter(.data$data_type == "train"),
```
Comment (damonbayer) [inline hewr/R/process_state_forecast.R:15]: Now that we have sorted out a way to get eval data, we should probably stop having non-training data in the `train_dat`. Would require changes to https://github.com/CDCgov/pyrenew-hew/blob/main/pipelines/prep_data.py

File: hewr/R/process_state_forecast.R [hunk 2]
```diff
@@ -30,135 +113,119 @@ process_state_forecast <- function(model_run_dir,
...
+  )
 
   last_training_date <- combined_dat |>
-    dplyr::filter(data_type == "train") |>
-    dplyr::pull(date) |>
+    dplyr::filter(.data$data_type == "train") |>
+    dplyr::pull("date") |>
     max()
 
   other_ed_visits_samples <-
```
Comment (damonbayer) [inline hewr/R/process_state_forecast.R:134]: I know I wrote this, but I think this part could use a comment to explain that we are creating a fake posterior by replicating the true data.

**Review by damonbayer — APPROVED** [review]
Just a few suggestions. Please address directly or make issues before merging.

File: hewr/R/process_state_forecast.R (hunk 1 above)
Comment (dylanhmorris) [reply hewr/R/process_state_forecast.R:15]: Created #207

File: hewr/R/process_state_forecast.R (hunk 2 above)
Comment (dylanhmorris) [reply hewr/R/process_state_forecast.R]: done in e1ff3d7

**Pull request merged**